src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerUtils.java

```diff
@@ -86,7 +86,7 @@ public static void finalizeRegion(final AssemblyRegion region,
             if ( ! clippedRead.isEmpty() && clippedRead.getCigar().getReadLength() > 0 ) {
                 clippedRead = ReadClipper.hardClipToRegion( clippedRead, region.getExtendedSpan().getStart(), region.getExtendedSpan().getEnd() );
                 if ( region.readOverlapsRegion(clippedRead) && clippedRead.getLength() > 0 ) {
-                    readsToUse.add(clippedRead);
+                    readsToUse.add((clippedRead == myRead) ? clippedRead.deepCopy() : clippedRead);
```

Comment:
I wonder if we can get away with a less-expensive shallow copy (ie., `copy()`) rather than a costlier `deepCopy()` here?

Also, can you add a unit test that fails before the fix and passes after it?